src/Angor/Avalonia/Angor.Contexts.Funding/Founder/Operations/GetInvestments.cs

```diff
@@ -42,35 +43,126 @@ private async Task<Result<IEnumerable<Investment>>> GetInvestments(GetInvestment
         {
-            var combineInvestmentsAndApprovals = messages.Select(message =>
-            {
-                return nostrDecrypter.Decrypt(walletId, project.Id, message)
-                    .MapTry(serializer.Deserialize<SignRecoveryRequest>)
-                    .Map(request =>
-                    {
-                        var isApproved = approvals.Any(a => a.EventIdentifier == message.Id);
-                        return new Investment(message.Created, GetAmount(request.InvestmentTransactionHex), request.InvestmentTransactionHex, message.InvestorNostrPubKey, message.Id, isApproved);;
-                    });
-            }).CombineInOrder(", ");
+            return alreadyInvested.Any(investment => investment.TransactionId == transactionId);
+        }
+
+        private static bool IsApproved(string eventId, IEnumerable<ApprovalMessage> approvals)
+        {
+            return approvals.Any(approval => approval.EventIdentifier == eventId);
+        }
+        
+        private Investment CreateInvestment(
+            InvestmentRequest signRequest, 
+            IEnumerable<ApprovalMessage> approvals, 
+            List<ProjectInvestment> alreadyInvested)
+        {
+            var transaction = networkConfiguration.GetNetwork().CreateTransaction(signRequest.InvestmentTransactionHex);
+            var amount = GetAmount(transaction);
+            var transactionId = transaction.GetHash().ToString();
+    
+            var investmentStatus = DetermineInvestmentStatus(
+                signRequest.EventId, 
+                transactionId, 
+                approvals, 
+                alreadyInvested);
+    
+            return new Investment(
+                signRequest.EventId, 
+                signRequest.CreatedOn, 
+                signRequest.InvestmentTransactionHex, 
+                signRequest.InvestorNostrPubKey, 
+                amount, 
+                investmentStatus);
+        }
+
+        
+        private Task<Result<(IEnumerable<InvestmentRequest> requests, IEnumerable<ApprovalMessage> approvals, List<ProjectInvestment> alreadyInvested)>> 
+            GetInvestmentData(GetInvestmentsRequest request, string nostrPubKey)
+        {
+            return
+                from requests in Requests(request, nostrPubKey)
+                from approvals in Approvals(nostrPubKey)
+                from alreadyInvested in AlreadyInvested(request)
+                select (requests, approvals, alreadyInvested);
+        }
+
+        private Task<Result<List<ProjectInvestment>>> AlreadyInvested(GetInvestmentsRequest request)
+        {
+            return Result.Try(() => indexerService.GetInvestmentsAsync(request.ProjectId.Value));
+        }
 
-            var result = await combineInvestmentsAndApprovals;
-            return result;
+        private async Task<Result<IEnumerable<InvestmentRequest>>> Requests(GetInvestmentsRequest request, string nostrPubKey)
```

Comment:
This method does a lot of heavy lifting the name should reflect the work done, perhaps lookup remote requests or something to make it clear it's not a container.